File: Programmers/lv2/12924.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(int n) {
    int answer = 0;
    int temp = 1, pivot = 1, pivot2 = 1;
    
    while(pivot <= n) {
        if (temp == n) {
            answer++;
        }

        if (temp >= n) {
            pivot++;
            pivot2 = pivot;
            temp = pivot2;
        } else {
            temp += pivot2 + 1;
            pivot2++;
        }
    }

    return answer;
}

int main() {
    cout << solution(15);
    return 0;
}

File: Python/programmers2-3/12924.py
def solution(n):
    answer = 0
    
    for start_point in range(1, n+1):
        su = 0
        for point in range(start_point, n+2):
            if su < n:
                su += point
            elif su == n:
                answer += 1
                break
            else:
                break
    
    return answer